### PR TITLE
fix variables schema dump: `$defs`  -> `definitions`

### PR DESCRIPTION
**Pull request by zzstoatzz**
this PR applies the same workaround for dumping json schemas as we're currently using for `Block`, where we use swap `$defs` for `definitions` because that's the format the UI is expecting

### CODE REVIEW

File: src/integrations/prefect-aws/prefect_aws/workers/ecs_worker.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ class ECSJobConfiguration(BaseJobConfiguration):
     Job configuration for an ECS worker.
     """
 
-    aws_credentials: Optional[AwsCredentials] = Field(default_factory=AwsCredentials)
+    aws_credentials: Optional[AwsCredentials] = Field(None)
```
Comment (desertaxle) [inline src/integrations/prefect-aws/prefect_aws/workers/ecs_worker.py:264]: Why did you remove the `default_factory` here?

Comment (zzstoatzz) [reply src/integrations/prefect-aws/prefect_aws/workers/ecs_worker.py:264]: hmm I guess I was thinking the default factory seemed not so helpful, but now im realizing it probably scoops env via boto on the worker?

Comment (zzstoatzz) [reply src/integrations/prefect-aws/prefect_aws/workers/ecs_worker.py]: restored in [d25d6c6](https://github.com/PrefectHQ/prefect/pull/13971/commits/d25d6c647a232ed2c3ea50d3ec9fe7754d388e37)

**Review by desertaxle — APPROVED** [review]
